### ISSUE TITLE
OrderBy and Limit - wrong order?

### ISSUE THREAD

**Issue by andreashannig** [orientechnologies/orientdb#5153]
In the GratefulDeadConcerts database (Version 2.1.4) the query
```
select expand(out('sung_by')) from v order by name desc
```
calculates the result (1):

rid | name
------------ | -------------
# 9:156 | Weir_Hart
# 9:61 | Weir
... | ...

When adding a limit 
```
select expand(out('sung_by')) from v order by name desc *limit 2*
```
i expected to receive the first two records from (1). Instead the query returns

rid | name
------------ | -------------
# 9:74 | Spencer_Davis
# 9:8 | Garcia

Is this a bug or is there an error in the query?

**Comment by luigidellaquila** [comment]
Hi @andreashannig 

which OrientDB version are you using? 
I fixed a similar issue a few days ago, could you please try again with latest release?

Thanks

Luigi

**Comment by andreashannig** [comment]
Hi @luigidellaquila,

i'm using version 2.1.4 which i downloaded from the orientdb homepage this morning.

(I found the problem in my current application and i wanted to check, if i can reproduce the issue in the GratefulDeadConcerts database) 

Best,
Andreas

**Comment by luigidellaquila** [comment]
Hi @andreashannig 

you're right, my bad, I fixed it after 2.1.4 was released, so it's still in snapshot.
If you have a chance to build from sources, you can try it on your won. Anyway I'll do it here

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @andreashannig 

no way, it was a different issue, I have to verify it in depth.
In the meantime, as a work-around, you can use this:
```sql
select from (select expand(out('sung_by')) from v order by name desc) limit 2
```

Thanks

Luigi

**Comment by andreashannig** [comment]
Hi @luigidellaquila,

many thanks for your quick response, - and thank you for the workaround.

Best regards
Andreas

**Comment by luigidellaquila** [comment]
Hi @andreashannig 

it's fixed in 2.1.x branch, now I'm porting it to develop (2.2)

Thanks

Luigi

**Comment by luigidellaquila** [comment]
fixed also in develop branch.
Closing, please reopen if I missed something

Thanks

Luigi

**Comment by albertotonon** [comment]
Hello everybody! 
just FYI, I've also found a weird behaviour which I think is related to what @andreashannig  reported (copying from http://stackoverflow.com/questions/33239118/orientdb-weird-behaviour-of-limit?noredirect=1#comment54283435_33239118)

the query `select expand(in()) from Employee` returns `0 item(s) found. Query executed in 0.0 sec(s).`, while the query `select expand(in()) from Employee limit 1000` returns `52 item(s) found. Query executed in 0.062 sec(s).`

Unfortunately I cannot share the data I'm using, however, my Employee class has > 15k instances of which only ~1400 have an in-edge.

**Comment by luigidellaquila** [comment]
Hi @albertotonon 

you have two problems here: the first one was a wrong behavior of expand() operator when combined with limit (fixed in 2.1.x branch), the second one is that you are probably executing the query from Studio, that automatically applies a LIMIT 20. Try to change this default from the lower right corner of the query panel, setting it to 1000, you will probably see the same results
